File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.cpp

class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int val = INT_MAX;
        int num;

        for (int i = 0; i < nums.size(); i++) {
            num = nums[i];
            if (abs(num) < abs(val)) {
                val = num;
            } else if (abs(num) == abs(val)) {
                val = max(num, val);
            }
        }

        return val;
    }
    
};

File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        ret: int = -100000000
        for num in nums:
            if abs(num) < abs(ret):
                ret = num
            elif abs(ret) == abs(num):
                ret = max(ret, num)
        return ret